### DISCUSSION TITLE
Configurable dashboards

### DISCUSSION THREAD

**Discussion by adrianthedev** [avo-hq/avo#380] [Feature requests]
A few gems that help with metrics and data aggregation:

- https://github.com/asross/repor
- https://github.com/codesnik/calculate-all

**Comment by TheFehr** [comment]
Just a suggestion for graphs:
- https://github.com/ankane/chartkick

**Reply by adrianthedev** [reply]
This is just 🔥 Perfect for our use case. Thanks!

**Comment by adrianthedev** [comment]
We started [work](https://github.com/avo-hq/avo/pull/566) on this using Chartkick.

**Reply by luonet** [reply]
Great! Looking forward to this new feature!

**Reply by adrianthedev** [reply]
That's great!

We posted the initial documentation for this feature [here](https://docs.avohq.io/1.0/dashboards.html). Can you have a look and share some feedback? What questions come to your mind?

**Reply by luonet** [reply]
Thanks. I'll try.

**Reply by luonet** [reply]
1. Naming.

I noticed the Dashboard class had attributes id, name, description, grid_cols, while Card class had id, label ,description, cols. Is it meaningful to differentiate the `name` and `label`, `grid_cols` and `cols`? Or just use the same `name` and `cols` to keep consistency and eliminate extra mind burden?

2. Card select.

I noticed the Card had `ranges` and `range` attributes, which would be used as a condition select. But I think it's not flexible enough. What should I do if I want two select, one for the time start and another for the time end? It seems impossible in the current design. I think the framework could provide the best default options, but should also be flexible enough to extend to fit the extra use cases.

Thanks for your great work! @adrianthedev

**Comment by yorch** [comment]
Great to hear!!

Not sure if you are still looking to use `repor`, but was actually looking at this package a few weeks ago and there is a fork that is still maintained as opposed to the original library: https://github.com/chaunce/active_reporter

**Reply by adrianthedev** [reply]
Actually `chartkick` kinda checks a lot of boxes when it comed to charts. With Avo you will be able to leverage chartkick or bring your own without issues.

**Reply by yorch** [reply]
That's great. `active_reporter` is for querying the data as opposed to charting, so might be complimentary.

**Reply by adrianthedev** [reply]
Oh yeah. `active_reporter` looks perfect for querying the data.

The approach with Avo dashboards is that we don't really care where you get the data. You can pass it an array, a hash, or an `active_reporter` result. So you could use that for generating the data.

Thank you for the lead. I'll make sure to add it to the docs.
